jupyterlab_git/git.py

```diff
@@ -298,11 +302,25 @@ async def status(self, current_path):
         """
         Execute git status command & return the result.
         """
-        cmd = ["git", "status", "--porcelain", "-u", "-z"]
-        code, my_output, my_error = await execute(
-            cmd,
-            cwd=os.path.join(self.root_dir, current_path),
-        )
+        cwd = os.path.join(self.root_dir, current_path)
+        # Start by fetching to get accurate ahead/behind status
+        cmd = [
+            "git",
+            "fetch",
+            "--all",
+            "--prune",
+        ]  # Run prune by default to help beginners
```

Comment:
Will this slow things down? On my laptop with a reasonably fast internet connection (fast for america :cry: ) I get:
`time git fetch --prune --all`
gives:
```
Fetching origin

real	0m0.607s
user	0m0.028s
sys	0m0.034s
```

Reply:
It will definitely depend on the internet connection and the number of changes to be downloaded.

From experience in VS Code, I never got the impression things got stuck. So I propose to move forward with it. And if users starts to complain, we could move it in its own poll running parallel to the status refresh with a dedicated interval. What do you think?

Reply:
At the end I followed your advice and set up a separate poll to fetch remote. This has the side-effect of stopping that poll if the repository does not have remote defined.